inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_request_internal.hpp

```diff
@@ -228,13 +228,29 @@ class InferRequestInternal : virtual public IInferRequestInternal {
      * @param serial Whether to use multiple threads to execute the step
      */
     void execDataPreprocessing(InferenceEngine::BlobMap& inputs, bool serial = false) {
-        for (auto& input : inputs) {
-            // If there is a pre-process entry for an input then it must be pre-processed
-            // using preconfigured resize algorithm.
-            auto it = _preProcData.find(input.first);
-            if (it != _preProcData.end()) {
-                _preProcData[input.first]->execute(input.second, _networkInputs[input.first]->getPreProcess(), serial,
-                                                   m_curBatch);
+        for (auto& pp_result : inputs) {
+            auto const& input_name    = pp_result.first;
+            auto & result_blob        = pp_result.second;
+            auto const& ir_input_blob = _inputs[input_name];
+
+            auto                   pp_it  = _preProcData.find(input_name);
+            const bool preprocDataExists  = (pp_it != _preProcData.end());
+
+            Blob::Ptr roi_blob = (preprocDataExists) ? (*pp_it).second ->getRoiBlob() : Blob::Ptr{nullptr};
```

Comment:
Does `getRoiBlob` always returns non-NULL? Just from its name I would expect that this is an optional `crop` pre-processing, which might not be enabled by user (for example, only color space conversion preprocessing).

Reply:
I think yes, if `preprocDataExists` is `true`, it means that preporoc engine is created for particular input and `setRoiBlob` is called in `SetBlob`

Reply:
@ilya-lavrenov  is correct. 

`getRoiBlob` is a bad name and now should be something like `getUserBlob`. I have added a FIXME to rename it (and accompanying one `seRoiBlob` )